src/components/adaptive-cards/AdaptiveCard/AdaptiveCard.jsx

```diff
@@ -85,6 +86,31 @@ export default function AdaptiveCard({
   const data = templateInstance.expand({
     $root: context,
   });
+
+  const [elementsProps, SetElementsProps] = useState({});
+
+  const setElementProps = useCallback((element) => {
```

Comment:
the idea of having a generic place to put element props is good, but I would prefer more semantic function names. For example, let's have `setIsVisible` and `getIsVisible`. We'll add more semantic functions if needed. You can also define a `toggleIsVisible` if it helps